mpas_analysis/shared/generalized_reader/generalized_reader.py

```diff
@@ -124,9 +130,35 @@ def open_multifile_dataset(fileNames, calendar, simulationStartTime=None,
                                  startDate=startDate,
                                  endDate=endDate)
 
-    ds = xarray.open_mfdataset(fileNames,
-                               preprocess=preprocess_partial,
-                               decode_times=False, concat_dim='Time', **kwargs)
+    # we don't want to decode times, since we're typically using dates
+    # outside the allowed range
+    kwargs['decode_times'] = False
+    # files differ along the Time dimension
+    kwargs['concat_dim'] = 'Time'
+
+    autoclose = ((autocloseFileCount is not None) and
+                 len(fileNames) > autocloseFileCount)
+
+    openedWithAutoclose = False
+    if autoclose:
+        try:
+            ds = xarray.open_mfdataset(fileNames,
+                                       preprocess=preprocess_partial,
+                                       autoclose=True, **kwargs)
+            openedWithAutoclose = True
+        except TypeError as e:
+            if 'autoclose' in str(e):
+                # This indicates that xarray version doesn't support autoclose
```

Comment:
I don't think this exception handling is dependent on the xarray PR in any way unless you change the name of the argument.  If the argument is not supported (e.g. in the current version of xarray) you get a `TypeError`, which I have already verified is the case.